src/main/java/com/wire/helium/API.java

```diff
@@ -618,6 +628,273 @@ public User getSelf() {
+                Logger.error("isMlsEnabled - Public Keys error: %s, status: %d", msgError, featureConfigsResponse.getStatus());
+                return false;
+            }
+
+            try {
+                PublicKeysResponse publicKeysResponse = mlsPublicKeysResponse.readEntity(PublicKeysResponse.class);
+
+            } catch (Exception e) {
+                Logger.error("isMlsEnabled - Public Keys Deserialization error: %s", e.getMessage());
+                return false;
+            }
+
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * <p>
+     *     To upload client public key we PUT a {@link ClientUpdate} object containing the public keys
+     *     to /clients/{clientId}
+     * </p>
+     * <p>
+     *     As there is no return, in case it fails we just map the HTTP Code and log the message.
+     * </p>
+     *
+     * @param clientId      clientId to upload the public keys
+     * @param clientUpdate  the public keys
+     */
+    @Override
+    public void uploadClientPublicKey(String clientId, ClientUpdate clientUpdate) {
+        try {
+            Response response = clientsPath
+                .path(clientId)
+                .request(MediaType.APPLICATION_JSON)
+                .header(HttpHeaders.AUTHORIZATION, bearer(token))
+                .put(Entity.json(clientUpdate));
+
+            if (response.getStatus() >= 400) {
+                Logger.error("uploadClientPublicKey error, bad-request");
+            } else if (response.getStatus() == 200) {
+                Logger.info("uploadClientPublicKey success");
+            }
+        } catch (Exception e) {
+            Logger.error("uploadClientPublicKey error: %s", e.getMessage());
+        }
+    }
+
+    /**
+     * <p>
+     *     To upload client key packages we POST a {@link KeyPackageUpdate} object containing a list of package keys
+     *     to /mls/key-packages/self/{clientId}
+     * </p>
+     * <p>
+     *     As there is no return, in case it fails we just map the HTTP Code and log the message.
+     * </p>
+     *
+     * @param clientId          clientId to upload the package keys
+     * @param keyPackageUpdate  list of package keys
+     */
+    @Override
+    public void uploadClientKeyPackages(String clientId, KeyPackageUpdate keyPackageUpdate) {
+        try {
+            Response response = mlsPath
+                .path("key-packages")
+                .path("self")
+                .path(clientId)
+                .request(MediaType.APPLICATION_JSON)
+                .header(HttpHeaders.AUTHORIZATION, bearer(token))
+                .post(Entity.json(keyPackageUpdate));
+
+            if (response.getStatus() == 403) {
+                Logger.error("uploadClientKeyPackages error, mls-identity-mismatch");
+            } else if (response.getStatus() == 400) {
+                Logger.error("uploadClientKeyPackages error, mls-protocol-error");
+            } else if (response.getStatus() == 200) {
```

Comment:
Here the backend replies with 201 and not 200.
In other frameworks/libraries usually there is a HttpStatus.isSuccessful() method.
Now I've just made some research and in our tech stack there is
```
if(Response.Status.Family.familyOf(responseCode).equals(Response.Status.Family.SUCCESSFUL)){
    //do your thing
}
```

This will cover 200, 201, 202, 204.
I usually like using this helpers, because you avoid this issues of havin to look is success is 200 or 201 (and the same can be done for errors, if you have same behaviour for all 4xx and 5xx responses).

However it is perfectly fine to be restrictive and change this to 201 (you have to check the others also)